### PR TITLE
Speed up raft leader election

### PR DESCRIPTION
**Pull request by matthiashanel**
Skip wait if node was previous leader
Shorten the response time to minElectionTimeout

I noticed that I used the wrong timer. election timer purpose is that nodes avoid election at the same time and fight unnecessarily. Thus bad as a timeout. 

Signed-off-by: Matthias Hanel <mh@synadia.com>

All these changes remain valid with Dereks pr.

### CODE REVIEW

File: server/raft.go [hunk 1]
```diff
@@ -2069,23 +2079,28 @@ func (n *raft) runAsCandidate() {
 		case <-n.quit:
 			return
 		case <-elect.C:
+			n.switchToCandidate()
+			return
+		case <-tikChan:
+			// disable timeout and receipt of more votes
+			voteChan = nil
+			tikChan = nil
```
Comment (kozlovic) [inline server/raft.go:2087]: tikChan is time.After() and will fire only once, not sure setting it to nil here does anything, but doesn't hurt.

**Review by kozlovic — APPROVED** [review]
LGTM

**Review by derekcollison — COMMENTED** [review]
The better timer makes sense, but the last term I do not think is possible. That means a leader stepped down for some reason and then won the election again, which does not make any sense. In any other scenario being a leader in the last term is not possible.

File: server/raft.go (hunk 1 above)
Comment (matthiashanel) [reply server/raft.go]: remove

**Comment by matthiashanel** [comment]
> The better timer makes sense, but the last term I do not think is possible. That means a leader stepped down for some reason and then won the election again, which does not make any sense. In any other scenario being a leader in the last term is not possible.

@derekcollison , removed the code

**Review by kozlovic — APPROVED** [review]
LGTM

**Comment by derekcollison** [comment]
I changed the logic around this in the updates branch. Let's hold on this one for now.

**Pull request closed without merging**